WorkloadManagementSystem/Agent/SiteDirector.py

```diff
@@ -1216,67 +1226,20 @@ def _writePilotScript(self, workingDirectory, pilotOptions,
     return _writePilotWrapperFile(workingDirectory=workingDirectory, localPilot=localPilot)
 
   def updatePilotStatus(self):
-    """ Update status of pilots in transient states
+    """ Update status of pilots in transient and final states
     """
-    for queue in self.queueDict:
-      ce = self.queueDict[queue]['CE']
-      ceName = self.queueDict[queue]['CEName']
-      queueName = self.queueDict[queue]['QueueName']
-      ceType = self.queueDict[queue]['CEType']
-      siteName = self.queueDict[queue]['Site']
-
-      result = pilotAgentsDB.selectPilots({'DestinationSite': ceName,
-                                           'Queue': queueName,
-                                           'GridType': ceType,
-                                           'GridSite': siteName,
-                                           'Status': TRANSIENT_PILOT_STATUS,
-                                           'OwnerDN': self.pilotDN,
-                                           'OwnerGroup': self.pilotGroup})
-      if not result['OK']:
-        self.log.error('Failed to select pilots", ": %s' % result['Message'])
-        continue
-      pilotRefs = result['Value']
-      if not pilotRefs:
-        continue
-
-      result = pilotAgentsDB.getPilotInfo(pilotRefs)
-      if not result['OK']:
-        self.log.error('Failed to get pilots info from DB', result['Message'])
-        continue
-      pilotDict = result['Value']
-
-      stampedPilotRefs = []
-      for pRef in pilotDict:
-        if pilotDict[pRef]['PilotStamp']:
-          stampedPilotRefs.append(pRef + ":::" + pilotDict[pRef]['PilotStamp'])
-        else:
-          stampedPilotRefs = list(pilotRefs)
-          break
-
-      # This proxy is used for checking the pilot status and renewals
-      # We really need at least a few hours otherwise the renewed
-      # proxy may expire before we check again...
-      result = ce.isProxyValid(3 * 3600)
-      if not result['OK']:
-        result = gProxyManager.getPilotProxyFromDIRACGroup(self.pilotDN, self.pilotGroup, 23400)
-        if not result['OK']:
+    # Getting the status of of pilots in a queue implies the use of remote CEs and may lead to network latency
+    # Threads aim at overcoming such issues and thus 1 thread per queue is created to submit pilots
```

Comment:
IIUC what's threaded is the update but not the submission, am I wrong?

Reply:
Oops yes, I copied a comment from a method that I removed afterwards. 
I've changed the comment.